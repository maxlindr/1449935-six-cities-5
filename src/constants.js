export const OfferCardType = {
  FAVORITE: `favorite`,
  NEAR_PLACE: `near_place`,
  CITIES: `cities`
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
  PENDING: `PENDING`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVORITES: `/favorites`,
  ERROR: `/error`,
  OFFER: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};

export const ErrorMessage = {
  NOT_FOUND: `Resource not found`,
  GENERAL: `Oops! Something went wrong`,
  UNAUTHORIZED: `Unauthorized`,
};
