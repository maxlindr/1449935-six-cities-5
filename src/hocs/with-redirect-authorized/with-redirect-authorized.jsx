import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../store/selectors';
import {Redirect} from 'react-router-dom';
import {AppRoute, AuthorizationStatus} from '../../constants';
import {omitProperties} from '../../utils';

const withRedirectAuthorized = (WrappedComponent) => {
  const WithRedirectAuthorized = (props) => (
    props.isAuthorized ? (
      <Redirect to={AppRoute.ROOT} />
    ) : (
      <WrappedComponent {...omitProperties(props, [`isAuthorized`])} />
    )
  );

  WithRedirectAuthorized.propTypes = {
    isAuthorized: PropTypes.bool.isRequired,
  };

  return WithRedirectAuthorized;
};

const mapStateToProps = (state) => ({
  isAuthorized: getAuthorizationStatus(state) === AuthorizationStatus.AUTHORIZED,
});

export {withRedirectAuthorized};

export default (WrapperComponent) => connect(mapStateToProps)(
    withRedirectAuthorized(WrapperComponent)
);
