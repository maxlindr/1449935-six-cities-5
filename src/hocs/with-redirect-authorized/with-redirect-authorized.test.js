import React from 'react';
import {MemoryRouter, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {withRedirectAuthorized} from './with-redirect-authorized';
import {AppRoute} from '../../constants';

const WrappedComponent = withRedirectAuthorized(() => <div>WrappedComponent component</div>);
const RedirectComponent = () => <div>Redirect component</div>;

const renderComponent = (isAuthorized) =>
  renderer.create(
      <MemoryRouter initialEntries={[AppRoute.LOGIN]} initialIndex={0}>
        <Route path={AppRoute.LOGIN} exact>
          <WrappedComponent isAuthorized={isAuthorized} />
        </Route>

        <Route path={AppRoute.ROOT} exact>
          <RedirectComponent />
        </Route>
      </MemoryRouter>
  );

describe(`withRedirectAuthorized`, () => {
  it(`должен корректно отображаться если проп isAuthorized=false`, () => {
    const tree = renderComponent(false).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`должен редиректить на маршрут '/' если isAuthorized=true`, () => {
    const tree = renderComponent(true).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
