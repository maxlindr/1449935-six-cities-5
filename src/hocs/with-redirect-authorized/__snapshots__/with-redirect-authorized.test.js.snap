// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRedirectAuthorized должен корректно отображаться если проп isAuthorized=false 1`] = `
<div>
  WrappedComponent component
</div>
`;

exports[`withRedirectAuthorized должен редиректить на маршрут '/' если isAuthorized=true 1`] = `
<div>
  Redirect component
</div>
`;
